### DISCUSSION TITLE
Global `colored` option

### DISCUSSION THREAD

**Discussion by fitrh** [nvim-lualine/lualine.nvim#596] [Ideas]
Once #566 merged, it seems easier for me to customize the colors, thanks.

Here's how I configure lualine:

- Define `vim.g.statusline_monochrome`
- For each component, I use that variable to set `color` and other color related properties if any, e.g. `colored`, `diff.diff_color`

How about having a global `colored` option to enable or disable the entire statusline component colors ?
enabled means each component uses its color, disabled means using `StatusLine` group or we can pass another highlight group (*as string or table `{ fg = ..., bg = ..., ...` ?*) to be used as global statusline color.

I have the base implementation on my local setup but I think it's still poor.

Just post the idea here in case anyone is interested

**Comment by shadmansaleh** [comment]
I'm not sure if I understood you correctly but does setting colored = true / false in options table not work ?

```lua
require'lualine'.setup {
  options = {
     colored = false,
  }
}
```

**Reply by fitrh** [reply]
```
Global options can be used as local options (can be applied to specific
components) but you cannot use local options as global. Global option used
```

Thank you, that exactly what I want, did not read the documentation carefully

**Reply by shadmansaleh** [reply]
Those lines in docs are confusing . Basically all components inherit options provided in `options` table .

**Reply by fitrh** [reply]
Yes, but apparently not all components respect the local `colored` option, e.g. `branch` component

**Reply by shadmansaleh** [reply]
Are you asking for `color` option to respect `colored`?

**Reply by fitrh** [reply]
Isn't that how the `colored` option works ? like in the `diagnostics` and `diff` components

**Reply by shadmansaleh** [reply]
Currently `color` option doesn't check for `colored` though it's quite trivial to add .

**Reply by fitrh** [reply]
I see, I think because of the `diagnostic` and `diff` components, other components have that option too
